File: lc-dc-sep/lc-dc-sep13/solution.cpp
// Problem: 1310. XOR Queries of a Subarray

class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> res;
        int n = arr.size();
        for(int i=1; i<n; i++) arr[i] ^= arr[i-1];
        n = queries.size();
        for(auto& query: queries){
            if(query[0] > 0) res.push_back(arr[query[0]-1] ^ arr[query[1]]);
            else res.push_back(arr[query[1]]);
        }
        return res;
    }
};

File: lc-dc-sep/lc-dc-sep13/solution.py
# Problem: 1310. XOR Queries of a Subarray

class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        res = []
        n = len(arr)
        for i in range(1, n):
            arr[i] ^= arr[i-1]
        
        for l, r in queries:
            if l > 0:
                res.append(arr[l-1]^arr[r])
            else:
                res.append(arr[r])
        return res